mne/io/fiff/tests/test_raw.py

```diff
@@ -115,6 +115,33 @@ def test_copy_append():
     assert_equal(data.shape[1], 2 * raw._data.shape[1])
 
 
+@testing.requires_testing_data
+def test_append_channels():
+    """Test raw splitting / re-appending channel types
+    """
+    raw = Raw(fif_fname, preload=True)
```

Comment:
which reminds me, what happens if I try to channel-concatenate raws with different numbers of time points? There should be an error. I think right now it is cryptic, you should add a check to make sure it's informative.

what happens if I try to channel-concatenate raws with different sampling rates, but the same number of time points?

what happens if I try to channel-concatenate raws with `.info` with e.g. different subject names?

etc.